src/main/java/fi/otavanopisto/kuntaapi/server/integrations/ptv/PtvServiceLocationServiceChannelEntityUpdater.java

```diff
@@ -0,0 +1,144 @@
+import fi.otavanopisto.kuntaapi.server.cache.ModificationHashCache;
+import fi.otavanopisto.kuntaapi.server.controllers.IdentifierController;
+import fi.otavanopisto.kuntaapi.server.discover.EntityUpdater;
+import fi.otavanopisto.kuntaapi.server.id.IdController;
+import fi.otavanopisto.kuntaapi.server.id.OrganizationId;
+import fi.otavanopisto.kuntaapi.server.id.ServiceLocationServiceChannelId;
+import fi.otavanopisto.kuntaapi.server.integrations.KuntaApiConsts;
+import fi.otavanopisto.kuntaapi.server.integrations.KuntaApiIdFactory;
+import fi.otavanopisto.kuntaapi.server.integrations.ptv.resources.PtvServiceLocationServiceChannelResourceContainer;
+import fi.otavanopisto.kuntaapi.server.integrations.ptv.tasks.ServiceLocationServiceChannelIdTaskQueue;
+import fi.otavanopisto.kuntaapi.server.persistence.model.Identifier;
+import fi.otavanopisto.kuntaapi.server.settings.SystemSettingController;
+import fi.otavanopisto.kuntaapi.server.tasks.IdTask;
+import fi.otavanopisto.kuntaapi.server.tasks.IdTask.Operation;
+
+@ApplicationScoped
+@Singleton
+@AccessTimeout (unit = TimeUnit.HOURS, value = 1l)
+@SuppressWarnings ("squid:S3306")
+public class PtvServiceLocationServiceChannelEntityUpdater extends EntityUpdater {
+
+  @Inject
+  private Logger logger;
+
+  @Inject
+  private SystemSettingController systemSettingController;
+
+  @Inject
+  private PtvApi ptvApi;
+  
+  @Inject
+  private KuntaApiIdFactory kuntaApiIdFactory;
+
+  @Inject
+  private PtvIdFactory ptvIdFactory;
+  
+  @Inject
+  private PtvTranslator ptvTranslator;
+
+  @Inject
+  private IdController idController;
+
+  @Inject
+  private IdentifierController identifierController;
+
+  @Inject
+  private PtvServiceLocationServiceChannelResourceContainer ptvServiceLocationServiceChannelResourceContainer;
+
+  @Inject
+  private ModificationHashCache modificationHashCache;
+  
+  @Inject
+  private ServiceLocationServiceChannelIdTaskQueue serviceLocationServiceChannelIdTaskQueue;
+
+  @Resource
+  private TimerService timerService;
+
+  @Override
+  public String getName() {
+    return "ptv-service-location-channels";
+  }
+
+  @Override
+  public void timeout() {
+    executeNextTask();
+  }
+  
+  @Override
+  public TimerService getTimerService() {
+    return timerService;
+  }
+  
+  private void executeNextTask() {
+    IdTask<ServiceLocationServiceChannelId> task = serviceLocationServiceChannelIdTaskQueue.next();
+    if (task != null) {
+      ServiceLocationServiceChannelId serviceLocationServiceChannelId = task.getId();
+      
+      if (task.getOperation() == Operation.UPDATE) {
+        updateServiceChannelChannel(serviceLocationServiceChannelId, task.getOrderIndex());
+      } else if (task.getOperation() == Operation.REMOVE) {
+        deleteServiceChannelChannel(serviceLocationServiceChannelId);
+      }
+    }
+  }
+
+  private void updateServiceChannelChannel(ServiceLocationServiceChannelId ptvServiceLocationServiceChannelId, Long orderIndex) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this unused private "updateServiceChannelChannel" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)